### PR TITLE
doc: restructure user guides section

### PR DESCRIPTION
**Pull request by ru-fu**
The User guides section has grown a lot. Restructure it to make
it easier for users to find the required information.

Ref. NCSDK-5277

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/dm_glossary.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   A Git commit, including a unique SHA and a commit message.
+
+Patch
+   See Commit.
+
+Commit tag
+   A tag prepended to the first line of the commit message to ease filtering and identification of particular commit types.
+
+Pull Request
+   A GitHub Pull Request, a set of commits that are sent for code review using GitHub.
```
Comment (greg-fer) [inline doc/nrf/dm_glossary.rst:60]: I'd mention here "PR", as there are many occurrences of this form in docs.

File: doc/nrf/app_boards.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+The |NCS| provides board definitions for all Nordic Semiconductor devices.
+In addition, you can define custom boards.
+
+.. _gs_programming_board_names:
+
+Board names
+***********
+
+The following tables lists all boards and build targets for Nordic Semiconductor's hardware platforms.
```
Comment (b-gent) [inline doc/nrf/app_boards.rst:18]:
```suggestion
The following table lists all boards and build targets for Nordic Semiconductor's hardware platforms.
```
We need to ensure that we are always informed when a new build target is added, I understand this is not the case right now.

File: doc/nrf/app_build_system.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+   :depth: 2
+
+The |NCS| build and configuration system is based on the one from Zephyr, with some additions.
+
+Zephyr's build and configuration system
+***************************************
+
+Zephyr's build and configuration system uses the following building blocks as a foundation:
+
+* CMake, the cross-platform build system generator,
```
Comment (b-gent) [inline doc/nrf/app_build_system.rst:17]:
```suggestion
* CMake, the cross-platform build system generator.
```
I know this was here before but we can use this chance to correct these small things.

File: doc/nrf/app_build_system.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+* Kconfig, a powerful configuration system also used in the Linux kernel,
```
Comment (b-gent) [inline doc/nrf/app_build_system.rst:18]:
```suggestion
* Kconfig, a powerful configuration system also used in the Linux kernel.
```

File: doc/nrf/dm_adding_code.rst [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+If you have changes in additional repositories beyond `sdk-nrf`_ itself, you can point to your own forks of those in the :file:`west.yml` included in your fork.
+
+Workflow 4: Application as the manifest repository
+==================================================
+
+An additional possibility is to take advantage of west to manage your own set of repositories.
+This workflow is particularly beneficial if your application is split among multiple repositories or, just like in the previous workflow, if you want to make changes to one or more |NCS| repositories, since it allows you to define the full set of repositories yourself.
+
+In order to implement this approach you first need to create a manifest repository of your own, which just means a repository that contains a :file:`west.yml` manifest file in its root.
```
Comment (greg-fer) [inline doc/nrf/dm_adding_code.rst:78]: I would break this section into smaller subsections corresponding to steps, which I counted at least 5 here.

**Review by greg-fer — COMMENTED** [review]
I really like the changes and the new structure.
Some suggestions for you to consider.

File: doc/nrf/dm_code_base.rst [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+A :ref:`manifest repository <zephyr:west-manifests>`, `sdk-nrf`_, contains a file in its root folder, :file:`west.yml`, which lists all other repositories (west projects) included in the |NCS|.
+The |NCS| repository structure has a star topology, with the `sdk-nrf`_ repository being the center of the star and all other repositories being west projects that are managed by :file:`west.yml`.
+This is equivalent to topology T2 in the :ref:`west documentation <zephyr:west-multi-repo>`.
+
+.. figure:: images/ncs-west-repos.png
+   :alt: A graphical depiction of the |NCS| repository structure
+
+   The |NCS| repository structure
+
+The figure above depicts the |NCS| repository structure.
```
[the remark this answers is not in the document]
Comment (OisinFitz) [reply doc/nrf/dm_code_base.rst:80]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_code_base.rst [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+A central concept with this repository structure is that each revision (in Git terms) of the `sdk-nrf`_ repository completely determines the revisions of all other
+repositories (i.e. the west projects).
+This means that the linear Git history of this manifest repository also determines the history of the repository set in its entirety, thanks to the :file:`west.yml` `west manifest file`_ being part of the manifest repository.
+West reads the contents of the manifest file to find out which revisions of the project repositories are to be checked out every time ``west update`` is run.
```
[the remark this answers is not in the document]
Comment (OisinFitz) [reply doc/nrf/dm_code_base.rst:84]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_code_base.rst [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+In this way, you can decide to work with a specific |NCS| release either by initializing a new west installation at a particular tag or by checking out the corresponding tag for a release in an existing installation and then updating your project repositories to the corresponding state with ``west update``.
+Alternatively, you can work with the latest state of development by using the master branch of the `sdk-nrf`_ repository, updating it with Git regularly and using ``west update`` to update the project repositories every time the manifest repository changes.
```
[the remark this answers is not in the document]
Comment (OisinFitz) [reply doc/nrf/dm_code_base.rst:86]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_code_base.rst [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+Revisions
+*********
+
+There are two fundamental revisions that are relevant to most |NCS| users:
+
+* The ``master`` branch of the `sdk-nrf`_ repository
+* Any Git tag (i.e. release) of the `sdk-nrf`_ repository
+
+As discussed above, the revision of the manifest repository, `sdk-nrf`_, uniquely determines the revisions of all other repositories, so a discussion about |NCS| revisions can be essentially limited to the manifest repository revision.
+
```
[the remark this answers is not in the document]
Comment (OisinFitz) [reply doc/nrf/dm_code_base.rst:98]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_managing_code.rst [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+
+#. Create a `GitHub fork`_ using the :guilabel:`Fork` button in the GitHub user interface.
+#. Add the newly created remote repository as a Git remote::
+
+     cd ncs/{folder_path}
+     # Rename the default remote from 'origin' to 'ncs', if required
+     git remote rename origin ncs
+     git remote add origin https://github.com/{username}/{repo}.git
+
+   For example, to create a fork of the `sdk-nrf`_ repository for GitHub user ``foo``::
```
[the remark this answers is not in the document]
Comment (OisinFitz) [reply doc/nrf/dm_managing_code.rst:110]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_adding_code.rst [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+* The application(s) will require updates of the |NCS| revision.
+
+Workflow 1: Eschew Git and west
+===============================
+
+If you have your own version control tools, you might want to simply not use Git or west at all, and instead rely on your own toolset.
+In such case, you must obtain a copy of the |NCS| on your file system and then manage the source code of both the SDK and your application yourself.
+
+Since no downloadable packages of the |NCS| are currently available, the simplest path to obtain the source code is to follow the instructions in the :ref:`corresponding section <dm-wf-get-ncs>` of the documentation.
+This requires you to install Git and west, but you can then ignore them from that point onwards.
```
[the remark this answers is not in the document]
Comment (OisinFitz) [reply doc/nrf/dm_adding_code.rst:40]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_adding_code.rst [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+If you need to update the copy of the |NCS| you are working with, you can :ref:`obtain the source code <dm-wf-get-ncs>` again, or, if you have kept the original set of repositories, :ref:`update it instead <dm-wf-update-ncs>`.
+Once you have obtained a copy of the |NCS| source code, which is self-contained in a single folder, you can then proceed to manage that code in any way you see fit.
```
[the remark this answers is not in the document]
Comment (OisinFitz) [reply doc/nrf/dm_adding_code.rst:42]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_adding_code.rst (hunk 5 above)
Comment (OisinFitz) [reply doc/nrf/dm_adding_code.rst:78]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_glossary.rst (hunk 1 above)
Comment (OisinFitz) [reply doc/nrf/dm_glossary.rst:60]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_code_base.rst [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+A fork can be hosted on any server, including a public Git hosting site like `GitHub`_.
+It is, however, important to differentiate between the generic concept of a fork and GitHub's concept of a `GitHub fork`_.
+When you create a GitHub fork, GitHub copies the original repository and tags the downstream repository (the fork) with a flag that allows users to send pull requests from the fork to its upstream repository.
+GitHub also supports creating forks without linking them to the upstream respository.
+See the `GitHub documentation <GitHub duplicate_>`_ for information about how to do this.
+
+.. _dm-repo-types:
+
+Repository types
```
[the remark this answers is not in the document]
Comment (OisinFitz) [reply doc/nrf/dm_code_base.rst:36]: This content will get a complete rework later. So, leave as is for now.

File: doc/nrf/dm_code_base.rst [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+* `sdk-nrf`_: The main repository for Nordic-developed software.
+* `nrfxlib`_: A repository containing linkable libraries developed by Nordic.
+
+OSS repositories, on the other hand, are typically soft forks of an upstream open source project, which Nordic maintains in order to keep a small set of changes that do not belong, or have not been merged, to the upstream official open-source repository.
+For example:
+
+* `sdk-zephyr`_ is a soft fork (and therefore a downstream) of the upstream official `Zephyr repository`_.
+* `sdk-mcuboot`_ is a soft fork (and therefore a downstream) of the upstream official `MCUboot repository`_.
+
+Repository structure
```
[the remark this answers is not in the document]
Comment (OisinFitz) [reply doc/nrf/dm_code_base.rst:65]: This content will get a complete rework later. So, leave as is for now.

**Review by OisinFitz — APPROVED** [review]

**Review by b-gent — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
Very nice rework

**Pull request merged**